flytekit/core/workflow.py

```diff
@@ -195,6 +195,8 @@ def __init__(
         self._nodes: List[Node] = []
         self._output_bindings: List[_literal_models.Binding] = []
         self._docs = docs
+        # Create a map that holds the outputs of each node.
+        self._intermediate_node_outputs: Dict[Node, Dict[str, Promise]] = {GLOBAL_START_NODE: {}}
```

Comment:
I found this line may also have the node execution independency issue. If we execute the same node twice, this intermediate_node_output is not None before the second execution. Not sure if this would have any problem, maybe we can clear the state on the begining of execution_node?